src/GraphQL.Upload.AspNetCore/GraphQLUploadMiddleware.cs

```diff
@@ -19,20 +19,23 @@ public class GraphQLUploadMiddleware<TSchema>
         private readonly RequestDelegate _next;
         private readonly GraphQLUploadOptions _options;
         private readonly GraphQLUploadRequestDeserializer _requestDeserializer;
-
+        private readonly string _graphQLPath;
         public GraphQLUploadMiddleware(ILogger<GraphQLUploadMiddleware<TSchema>> logger, RequestDelegate next,
-            GraphQLUploadOptions options, GraphQLUploadRequestDeserializer requestDeserializer)
+            GraphQLUploadOptions options, PathString path, GraphQLUploadRequestDeserializer requestDeserializer)
         {
             _logger = logger;
             _next = next ?? throw new ArgumentNullException(nameof(next));
             _options = options;
             _requestDeserializer = requestDeserializer;
+            _graphQLPath = path;
         }
 
         public async Task InvokeAsync(HttpContext context)
         {
-            if (!context.Request.HasFormContentType)
+            if (!context.Request.HasFormContentType                 
+                || !context.Request.Path.Value.ToLower().StartsWith( _graphQLPath.ToLower() ))
```

Comment:
Please use the dedicated method provided by ASP.NET Core:
```suggestion
                || !context.Request.Path.StartsWithSegments(_graphQLPath))
```